query-engine/connector-test-kit-rs/query-tests-setup/src/runner/mod.rs

```diff
@@ -11,18 +13,42 @@ use query_core::{
 };
 use query_engine_metrics::MetricRegistry;
 use request_handlers::{
-    load_executor, BatchTransactionOption, ConnectorMode, GraphqlBody, JsonBatchQuery, JsonBody, JsonSingleQuery,
-    MultiQuery, RequestBody, RequestHandler,
+    BatchTransactionOption, ConnectorMode, GraphqlBody, JsonBatchQuery, JsonBody, JsonSingleQuery, MultiQuery,
+    RequestBody, RequestHandler,
+};
+use serde_json::json;
+use std::{
+    env,
+    sync::{atomic::AtomicUsize, Arc},
 };
-use std::{env, sync::Arc};
 
 pub type TxResult = Result<(), user_facing_errors::Error>;
 
 pub(crate) type Executor = Box<dyn QueryExecutor + Send + Sync>;
 
+pub enum RunnerExecutor {
+    Builtin(Executor),
+    External(usize),
+}
+
+impl RunnerExecutor {
+    async fn new_external(url: &str, schema: &str) -> TestResult<RunnerExecutor> {
```

Comment:
Guess this is what we could parameterize with a driver that we push down to the initializeSchema call sent to the node process. That one could grab the driver name and instantiate the driver provider the test kit depends on all existing drivers. Is that correct?

Reply:
The other direction I was imagining was one executor package / program per adapter we want to test, with the shared logic in a small package somewhere. But it would probably be more complicated than selecting at runtime.